app/scripts/modules/core/src/managed/StatusBubble.tsx

```diff
@@ -36,16 +38,47 @@ const transitionConfig = {
   config: { mass: 1, tension: 400, friction: 30 },
 };
 
-export const StatusBubble = memo(({ iconName, appearance, size }: IStatusBubbleProps) => {
+export const StatusBubble = memo(({ appearance, iconName, size, quantity }: IStatusBubbleProps) => {
   const transitions = useTransition(iconName, null, transitionConfig);
+  let quantityPill: ReactNode = null;
+  let withQuantity = false;
+
+  // Render quantity text only for size >= small.
+  if (quantity && (size === 'small' || size === 'medium' || size === 'large' || size === 'extraLarge')) {
+    withQuantity = true;
+    quantityPill = (
+      <span
+        className={classnames({
+          Pill: true,
+          [size]: true,
+          [`status-${appearance}`]: true,
+        })}
+      >
+        {quantity}
+      </span>
+    );
+  }
```

Comment:
_slightly_ subjective, but in my experience makes a meaningful impact on how code is reasoned about: i recommend moving this into a function outside the component (e.g. `getQuantityPill()`) as opposed to using `let` bindings with conditional assignment. occasionally `let` bindings are necessary — sometimes because there's no other solution to the problem, other times for performance reasons like avoiding multiple passes over an array. In most cases, though, opting for a function makes reasoning about the state you'll be in during runtime more straightforward.

if you do decide to refactor, i think tactically it would mean you'd do something like `const quantityPill = getQuantityPill(...);` and then use `!!quantityPill` where you were using `withQuantity`, and `getQuantityPill` would give you back an element or nothing.

Reply:
Totally agree. It is way more readable now!